src/cares_wrap.cc

```diff
@@ -525,6 +544,573 @@ class QueryWrap : public AsyncWrap {
   }
 };
 
+Local<Array> AddrTTLToArray(Environment* env,
+                            ares_addrttl* addrttls,
+                            int naddrttls) {
+  auto isolate = env->isolate();
+  EscapableHandleScope escapable_handle_scope(isolate);
+
+  Local<Array> ttls = Array::New(isolate, naddrttls);
+  auto context = env->context();
+  for (int i = 0; i < naddrttls; i += 1) {
+    auto value = Integer::New(isolate, addrttls[i].ttl);
+    ttls->Set(context, i, value).FromJust();
+  }
+
+  return escapable_handle_scope.Escape(ttls);
+}
+
+
+Local<Array> AddrTTL6ToArray(Environment* env,
+                             ares_addr6ttl* addrttls,
+                             int naddrttls) {
+  auto isolate = env->isolate();
+  EscapableHandleScope escapable_handle_scope(isolate);
+
+  Local<Array> ttls = Array::New(isolate, naddrttls);
+  auto context = env->context();
+  for (int i = 0; i < naddrttls; i += 1) {
+    auto value = Integer::New(isolate, addrttls[i].ttl);
+    ttls->Set(context, i, value).FromJust();
+  }
+
+  return escapable_handle_scope.Escape(ttls);
+}
+
+
+int ParseGeneralReply(Environment* env,
+                      unsigned char* buf,
+                      int len,
+                      int* type,
+                      Local<Array> ret,
+                      void* addrttls = nullptr,
+                      int* naddrttls = nullptr) {
+  HandleScope handle_scope(env->isolate());
+  hostent* host;
+
+  int status;
+  switch (*type) {
+  case ns_t_a:
+  case ns_t_cname:
+  case ns_t_cname_or_a:
+    status = ares_parse_a_reply(buf,
+                                len,
+                                &host,
+                                reinterpret_cast<ares_addrttl*>(addrttls),
+                                naddrttls);
+    break;
+  case ns_t_aaaa:
+    status = ares_parse_aaaa_reply(buf,
+                                   len,
+                                   &host,
+                                   reinterpret_cast<ares_addr6ttl*>(addrttls),
+                                   naddrttls);
+    break;
+  case ns_t_ns:
+    status = ares_parse_ns_reply(buf, len, &host);
+    break;
+  case ns_t_ptr:
+    status = ares_parse_ptr_reply(buf, len, NULL, 0, AF_INET, &host);
+    break;
+  default:
+    CHECK(0 && "Bad NS type");
+  }
+
+  if (status != ARES_SUCCESS)
+    return status;
+
+  /* If it's `CNAME`, return the CNAME value */
+  /* And if it's `CNAME_OR_A` and it has value in `h_name` and `h_aliases[0]` */
+  /* We consider it's a CNAME record, otherwise we consider it's an A record */
```

Comment:
This continues the sentence from the line before, right? Could you start it with a lowercase `w` then? And maybe consider folding the comments into a single multi-line comment, i.e.
```
/* foo
 * bar
 * baz */
```